osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -417,11 +433,8 @@ public void Export(TModel item)
         /// <param name="filename">An optional filename for the new file. Will use the previous filename if not specified.</param>
         public void ReplaceFile(TModel model, TFileModel file, Stream contents, string filename = null)
         {
-            using (ContextFactory.GetForWrite())
-            {
-                DeleteFile(model, file);
-                AddFile(model, contents, filename ?? file.Filename);
-            }
+            DeleteFile(model, file);
+            AddFile(model, contents, filename ?? file.Filename);
```

Comment:
why is this not under a write usage? operation is no longer atomic.

Reply:
Previous strategy of atomic operations causes efcore 5 tracking issue so it has to be decoupled from one transaction as a temporary workaround (since we are moving away from ef), see previous comments from @peppy:
```
If you are able to make it work with your strategy of a double pass, then that should be sufficient for now, given that we are hopefully moving away from EF in the future so this is all temporary.
```

Reply:
i'm not sure i get it. the comment that i read stated that the problem was due to some file entities already being possibly tracked. i'm not following why that leads to dropping consistency in this particular operation and potentially not restoring the deleted file in case the add fails.

Reply:
I think I found another way to fix tracking issues, I'm now working on it.

Reply:
Reverted.